### PR TITLE
Feature NonStackable Boxes on ClassNode Creation.

### PR DESCRIPTION
**Pull request by AaronWB**
Works on Save and Load.
Works on CLI to GUI, without closing application.
Works with adding a class moving it out of the way and its old spot is replaced when a new class is added.
Works and replaces when a class node is deleted.

### CODE REVIEW

File: UML editor/src/main/java/codecain/GraphicalUserInterface/Controller.java [hunk 1]
```diff
@@ -43,34 +43,35 @@ public class Controller {
...
+            return;
+        } else if (UMLClass.exists(className)) {
+            showAlert(AlertType.ERROR, "Error", "Class '" + className + "' already exists.", "");
+            return;
         }
+    
+        UMLClass.addClass(className);
+        UMLClassInfo classInfo = UMLClass.getClassInfo(className);
+        ClassNode classNode = new ClassNode(classInfo);
+
```
Comment (colbycrutcher) [inline UML editor/src/main/java/codecain/GraphicalUserInterface/Controller.java:63]: New way to calculate and set.

File: UML editor/src/test/java/CalculateAndSetPositionTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                    }
+                }
+            }
+        }
+
+        // Set calculated position
+        classNode.setLayoutX(x);
+        classNode.setLayoutY(y);
+
+        // Sync the calculated position with classInfo
```
Comment (colbycrutcher) [inline UML editor/src/test/java/CalculateAndSetPositionTest.java:70]: New tests look good to test calculated positions

File: UML editor/src/main/java/codecain/GraphicalUserInterface/Controller.java [hunk 3]
```diff
@@ -962,7 +977,71 @@ public void populateGUIFromClassMap() {
...
+        }
+        
+        boolean positionFound = false;
+        while (!positionFound) {
+            positionFound = true; 
+            for (Node node : container.getChildren()) {
+                if (node instanceof ClassNode) {
+                    double otherX = node.getLayoutX();
+                    double otherY = node.getLayoutY();
+                    if (Math.abs(x - otherX) < nodeWidth + padding && Math.abs(y - otherY) < nodeHeight + padding) {
```
Comment (colbycrutcher) [inline UML editor/src/main/java/codecain/GraphicalUserInterface/Controller.java:1025]: Looks good for class box positioning

**Review by colbycrutcher — APPROVED** [review]
Looks good. Cleans up some old code, and calculations look good from the outside.

File: UML editor/src/main/java/codecain/GraphicalUserInterface/Controller.java [hunk 4]
```diff
@@ -948,12 +949,26 @@ private void loadBtn() throws IOException {
...
-            classNode.setLayoutY(classInfo.getY());
+            if (classInfo.getX() == 0 && classInfo.getY() == 0) {
+                calculateAndSetPosition(classNode, classInfo, nodeContainer);
+            } else {
+                classNode.setLayoutX(classInfo.getX());
+                classNode.setLayoutY(classInfo.getY());
+            }
             classNode.setOnMouseClicked(event -> selectClassNode(classNode));
             nodeContainer.getChildren().add(classNode);
         });
```
Comment (rrudolfo) [inline UML editor/src/main/java/codecain/GraphicalUserInterface/Controller.java:974]: Good adjusted code to not allow overlapping class boxes.

File: UML editor/src/test/java/CalculateAndSetPositionTest.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        newNode.setPrefWidth(200);
+        newNode.setPrefHeight(300);
+
+        calculateAndSetPosition(newNode, newClassInfo, container);
+
+        assertEquals(220, newClassInfo.getX());
+        assertEquals(0, newClassInfo.getY());
+    }
+    
+}
```
Comment (rrudolfo) [inline UML editor/src/test/java/CalculateAndSetPositionTest.java:137]: Tests look good and the variety of tests is good as well.

**Review by rrudolfo — APPROVED** [review]
The overall changes look good and the test class tests a good amount of boundaries

**Pull request merged**